src/redturtle/gallery/browser/gallery.py

```diff
@@ -28,13 +38,16 @@ def __init__(self, context, request):
         self.getIndex()
 
     def getContents(self):
-        self.contents = self.context.listFolderContents(
-                            contentFilter={'portal_type': 'Image'}
-                        )
-        return self.contents
+        if ICollection.providedBy(self.context):
```

Comment:
If I read this correctly, this should be the other way around... from what I understand here, you should use `queryCatalog` when it is a Collection, and `listFolderContents` otherwise.